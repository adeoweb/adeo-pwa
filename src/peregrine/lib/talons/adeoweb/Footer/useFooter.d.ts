type TUseFooter = {
    copyrightText: string;
};

export function useFooter(): TUseFooter;
