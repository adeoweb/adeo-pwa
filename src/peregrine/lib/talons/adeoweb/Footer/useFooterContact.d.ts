type TUseuseFooterContact = {
    storeAddress: string;
    storePhone: string;
    storeEmail: string;
    storeWorkingHours: string;
};

export function useFooterContact(): TUseuseFooterContact;
