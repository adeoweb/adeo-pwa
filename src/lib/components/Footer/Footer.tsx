import React, { FunctionComponent } from 'react';
import ContactInformation from 'src/lib/components/Footer/ContactInformation/ContactInformation';
import FooterBottom from 'src/lib/components/Footer/FooterBottom';
import Newsletter from 'src/lib/components/Footer/Newsletter';
import AccountLinks from 'src/lib/components/Footer/AccountLinks';
import CmsBlockGroup from 'src/lib/components/CmsBlock';

const Footer: FunctionComponent = () => {
    return (
        <footer className="footer">
            <div className="footer-middle">
                <div className="container">
                    <div className="row">
                        <div className="col-lg-2 col-md-6">
                            <AccountLinks />
                        </div>
                        <div className="col-lg-3 col-md-6">
                            <CmsBlockGroup identifiers={'footer-menu-links'} />
                        </div>
                        <div className="col-lg-5 col-md-6">
                            <Newsletter />
                        </div>
                        <ContactInformation />
                    </div>
                </div>
            </div>
            <FooterBottom />
        </footer>
    );
};

export default Footer;
