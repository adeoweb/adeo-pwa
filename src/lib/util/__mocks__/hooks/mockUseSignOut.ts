import { TUseSignOut } from 'src/peregrine/lib/talons/adeoweb/SignOut/useSignOut';

const mock: TUseSignOut = { handleSignOut: () => null };

export default mock;
