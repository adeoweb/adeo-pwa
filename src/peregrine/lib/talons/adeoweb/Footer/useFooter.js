import { useConfigContext } from '../../../context/adeoweb/config';

export const useFooter = () => {
    const [{ copyright: copyrightText }] = useConfigContext();

    return {
        copyrightText: copyrightText || ''
    };
};
