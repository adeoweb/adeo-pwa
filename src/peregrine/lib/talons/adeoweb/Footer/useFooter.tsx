import { useConfigContext } from 'src/peregrine/lib/context/adeoweb/config';

type TUseFooter = {
    copyrightText: string;
};

export const useFooter = (): TUseFooter => {
    const [{ copyright: copyrightText = '' }] = useConfigContext();

    return {
        copyrightText
    };
};
