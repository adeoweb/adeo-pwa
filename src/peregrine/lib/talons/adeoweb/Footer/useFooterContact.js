import { useConfigContext } from '../../../context/adeoweb/config';

export const useFooterContact = () => {
    const [
        {
            store_address: storeAddress,
            store_phone: storePhone,
            store_email: storeEmail,
            store_working_hours: storeWorkingHours
        }
    ] = useConfigContext();

    return {
        storeAddress,
        storePhone,
        storeEmail,
        storeWorkingHours
    };
};
