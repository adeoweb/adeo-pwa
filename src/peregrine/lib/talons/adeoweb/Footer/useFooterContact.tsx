import { useConfigContext } from '../../../context/adeoweb/config';

type TUseFooterContact = {
    storeAddress: string;
    storePhone: string;
    storeEmail: string;
    storeWorkingHours: string;
};

export const useFooterContact = (): TUseFooterContact => {
    const [
        {
            store_address: storeAddress,
            store_phone: storePhone,
            store_email: storeEmail,
            store_working_hours: storeWorkingHours
        }
    ] = useConfigContext();

    return {
        storeAddress,
        storePhone,
        storeEmail,
        storeWorkingHours
    };
};
