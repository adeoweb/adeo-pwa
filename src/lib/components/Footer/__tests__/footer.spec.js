import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import testRenderer from 'react-test-renderer';

import mockUseSignOut from 'src/lib/util/__mocks__/hooks/mockUseSignOut';

import Footer from '../Footer';

jest.mock('owl.carousel', () => {});

jest.mock('src/peregrine/lib/talons/adeoweb/Footer/useFooter', () => {
    return {
        useFooter: () => ({
            copyrightText: 'Mocked Copyright Text'
        })
    };
});

jest.mock('@apollo/react-hooks', () => ({
    useQuery: jest.fn(() => {
        return {
            data: {
                cmsBlocks: {
                    items: ['']
                }
            }
        };
    })
}));

jest.mock('src/peregrine/lib/talons/adeoweb/Footer/useFooterContact', () => {
    return {
        useFooterContact: () => ({
            storeAddress: 'Mocked store address',
            storePhone: 'Mocked store phone',
            storeEmail: 'Mocked store email',
            storeWorkingHours: 'Mocked store workingHours'
        })
    };
});

jest.mock('src/peregrine/lib/context/adeoweb/user', () => {
    const userState = {
        isSignedIn: false
    };
    const userApi = {
        getUserDetails: jest.fn(),
        setToken: jest.fn()
    };
    const useUserContext = jest.fn(() => [userState, userApi]);

    return { useUserContext };
});

jest.mock('src/peregrine/lib/talons/adeoweb/SignOut/useSignOut', () => {
    return {
        useSignOut: () => mockUseSignOut
    };
});

test('Footer component renders correctly', () => {
    const component = testRenderer.create(
        <Router>
            <Footer />
        </Router>
    );
    expect(component.toJSON()).toMatchSnapshot();
});
