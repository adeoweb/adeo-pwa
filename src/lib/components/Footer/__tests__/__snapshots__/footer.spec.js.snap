// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<footer
  className="footer"
>
  <div
    className="footer-middle"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-2 col-md-6"
        >
          <div
            className="widget"
          >
            <h4
              className="widget-title"
            >
              My Account
            </h4>
            <ul
              className="links"
            >
              <li>
                <a
                  className="login-link"
                  href="/customer/login"
                  onClick={[Function]}
                >
                  Login
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="col-lg-3 col-md-6"
        >
          <div>
            <div />
          </div>
        </div>
        <div
          className="col-lg-5 col-md-6"
        >
          <div
            className="widget widget-newsletter"
          >
            <h4
              className="widget-title"
            >
              Subscribe newsletter
            </h4>
            <p>
              Get all the latest information on Events, Sales and Offers. Sign up for newsletter today.
            </p>
            <form
              className=""
              noValidate={true}
              onSubmit={[Function]}
            >
              <input
                className="form-control"
                name="email"
                onChange={[Function]}
                placeholder="Email address"
                type="text"
                value=""
              />
              <div
                className="invalid-feedback"
              />
              <button
                className="btn btn-primary"
                disabled={false}
                type="submit"
              >
                Subscribe
                <i
                  className="icon-angle-right"
                />
              </button>
            </form>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6"
        >
          <div
            className="widget"
          >
            <ul
              className="contact-info"
            >
              <li>
                <span
                  className="contact-info-label"
                >
                  Address
                  :
                </span>
                Mocked store address
              </li>
              <li>
                <span
                  className="contact-info-label"
                >
                  Phone
                  :
                </span>
                Toll Free
                 
                <a
                  href="tel:"
                >
                  Mocked store phone
                </a>
              </li>
              <li>
                <span
                  className="contact-info-label"
                >
                  Email
                  :
                </span>
                 
                <a
                  href="mailto:mail@example.com"
                >
                  Mocked store email
                </a>
              </li>
              <li>
                <span
                  className="contact-info-label"
                >
                  Working Days/Hours
                  :
                </span>
                Mocked store workingHours
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="footer-bottom"
    >
      <p
        className="footer-copyright"
      >
        Mocked Copyright Text
      </p>
      <a
        className="footer-adeoweb-signature"
        href="https://www.adeoweb.biz"
        rel="noopener"
        target="_blank"
      >
        <img
          alt="AdeoWeb"
          src="test-file-stub"
        />
        <span>
          ecommerce solutions
        </span>
      </a>
    </div>
  </div>
</footer>
`;
